cns/restserver/ipam.go

```diff
@@ -176,9 +171,12 @@ func (service *HTTPRestService) requestIPConfigsHandler(w http.ResponseWriter, r
 	if service.IPConfigsHandlerMiddleware != nil {
 		// Wrap the default datapath handlers with the middleware
 		wrappedHandler := service.IPConfigsHandlerMiddleware.IPConfigsRequestHandlerWrapper(service.requestIPConfigHandlerHelper, service.releaseIPConfigHandlerHelper)
-		ipConfigsResp, err = wrappedHandler(r.Context(), ipconfigsRequest)
+		ipConfigsResp, err = wrappedHandler(r.Context(), &ipconfigsRequest)
+		if reflect.DeepEqual(ipConfigsResp.PodIPInfo[0].HostPrimaryIPInfo, cns.HostIPInfo{}) || reflect.DeepEqual(ipConfigsResp.PodIPInfo[0].HostSecondaryIPInfo, cns.HostIPInfo{}) {
```

Comment:
I would advise against using `reflect` here. Can we just check whether `PrimaryIP` has been set or not?